src/status_im2/contexts/add_new_contact/views.cljs

```diff
@@ -0,0 +1,76 @@
+(ns status-im2.contexts.add-new-contact.views
+  (:require [status-im2.contexts.add-new-contact.style :as style]
+            [status-im.react-native.resources :as resources]
+            [i18n.i18n :as i18n]
+            [utils.debounce :as debounce]
+            [utils.re-frame :as rf]
+            [react-native.core :as rn]
+            [quo2.core :as quo]
+            [quo2.foundations.colors :as colors]))
+
+(defn new-contact
+  []
+  (let [{:keys [public-key state error]} (rf/sub [:contacts/new-identity])
+        error?                           (and (= state :error)
+                                              (= error :uncompressed-key))]
+    [:<>
+     [rn/image {:source (resources/get-image :add-new-contact)}]
+     [rn/touchable-opacity
+      {:style    style/close-button
+       :on-press #(rf/dispatch [:navigate-back])}
+      [quo/icon :i/close]]
+     [rn/view {:style style/content-outer-view}
+      [rn/view {:style style/content-inner-view}
+       [rn/view {:style style/text-combination-view}
+        [quo/text
+         {:size   :heading-1
+          :weight :semi-bold
+          :style  {:color colors/neutral-100}}
+         (i18n/label :t/add-a-contact)]
+        [quo/text
+         {:size   :paragraph-1
+          :weight :regular
+          :style  {:color colors/neutral-100}}
+         (i18n/label :t/find-your-friends)]]
+       [rn/view {:style style/key-view}
+        [rn/view {:style style/input-outer-view}
+         [quo/text
+          {:size   :paragraph-2
+           :weight :medium
+           :style  {:color         colors/neutral-50
+                    :margin-bottom 8}}
+          (i18n/label :t/ens-or-chat-key)]
+         [rn/text-input
+          {:style               (style/text-input error?)
+           :placeholder         (i18n/label :t/type-some-chat-key)
+           :accessibility-label :enter-contact-code-input
+           :auto-capitalize     :none
+           :multiline           true
+           :on-change-text      #(debounce/debounce-and-dispatch
+                                  [:contacts/set-new-identity %]
+                                  600)}]]
+        [rn/view {:style style/qr-button}
+         [rn/touchable-opacity {:on-press #(js/alert "TODO: to be implemented")}
+          [quo/icon :i/scan]]]]
+       [rn/view {:style style/info-message-view}
+        (when error?
+          [quo/icon :i/alert
+           {:size  16
+            :color (colors/theme-colors colors/danger-50 colors/white)}])
+        [quo/text
+         {:size  :paragraph-2
+          :align :left
+          :style {:color       colors/danger-50
+                  :margin-left 4}}
+         (when error? (i18n/label :t/not-a-chatkey))]]]
+      [rn/view {:style style/bottom-actions-view}
+       [quo/button
+        {:type                :primary
+         :size                40
+         :width               335
```

Comment:
layout should be flexible, we should never use height or width, only flex and margins, paddings